.github/actions/skupper-integration-tests/action.yml

```diff
@@ -0,0 +1,114 @@
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Test skupper-router main
+description: Execute skupper integration testing
+
+inputs:
+
+  qdrouterdImage:
+    required: true
+    description: Image to use in skupper integration tests
+
+  skupperGitRef:
+    required: false
+    description: Reference to skupper version
+    default: "0.8.7"
+
+runs:
+  using: "composite"
+  steps:
+
+      - name: Get Go cache paths
+        id: go-cache-paths
+        run: |
+          echo "::set-output name=go-build::$(go env GOCACHE)"
+          echo "::set-output name=go-mod::$(go env GOMODCACHE)"
+        shell: bash
+
+      - name: Install microk8s
+        run: |
+          sudo snap install microk8s --classic --channel=1.21/stable
+          sudo microk8s enable dns ingress storage registry:size=20Gi host-access:ip=10.0.1.1 metallb:10.64.140.43-10.64.140.49
+          sudo microk8s status --wait-ready
+        shell: bash
+
+      - name: Export microk8s config
+        if: "${{ always() }}"
+        run: |
+          mkdir -p $HOME/.kube
+          sudo microk8s config > $HOME/.kube/config
+        shell: bash
+
+      - name: Push image into microk8s
+        run: |
+          docker tag "${{ inputs.qdrouterdImage }}" localhost:32000/skupper-router:registry
+          timeout 5m bash -c 'until nc -z localhost 32000; do sleep 1; done' || :
+          docker push localhost:32000/skupper-router:registry
+          podman pull --tls-verify=false localhost:32000/skupper-router:registry
+        shell: bash
+
+      - name: Checkout Skupper
+        uses: actions/checkout@v2
+        with:
+          repository: 'skupperproject/skupper'
+          ref: "${{ inputs.skupperGitRef }}"
+          path: 'skupper'
+
+      - name: Restore Go cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.go-cache-paths.outputs.go-build }}
+            ${{ steps.go-cache-paths.outputs.go-mod }}
+          key: ${{ runner.os }}-go-build-${{ hashFiles('**/go.sum') }}
+
+      - name: Compile Skupper
+        run: |
+          make all
+          sudo install skupper /usr/local/bin
+        working-directory: skupper
+        shell: bash
+
+      - name: Run Skupper integration tests
+        run: |
+          export PUBLIC_1_INGRESS_HOST=10.0.1.1
+          export QDROUTERD_IMAGE=localhost:32000/skupper-router:registry
+
+          go test -count=1 -p=1 -timeout=60m -tags=integration -v ./test/integration/...
```

Comment:
you can remove `-p=1` here, which would speed things up

Reply:
-p=3 now, and it passed a few times in a row